### DISCUSSION TITLE
Custom Empires

### DISCUSSION THREAD

**Discussion by PrettyFlower** [Keriew/augustus#734] [General]
This feature allows map makers to make their own empires for scenarios. To create a custom empire, you'll need to make a custom_empires folder in your C3 base directory. Once that's created, you can start with the following example empire:
```XML
<?xml version="1.0"?>
<!DOCTYPE empire>
<empire version="1">
    <ornament type="The Stonehenge" />
    <ornament type="Gallic Wheat" />
    <border density="25">
        <edge x="169" y="215" hidden="true" />
        <edge x="142" y="157" />
        <edge x="181" y="126" />
        <edge x="217" y="128" />
        <edge x="233" y="110" />
        <edge x="312" y="71" />
        <edge x="325" y="78" />
        <edge x="344" y="70" />
        <edge x="370" y="84" hidden="true" />
        <edge x="454" y="126" />
        <edge x="496" y="282" />
        <edge x="466" y="359" />
        <edge x="474" y="425" />
        <edge x="465" y="433" />
        <edge x="440" y="419" />
        <edge x="409" y="444" />
        <edge x="402" y="469" />
        <edge x="262" y="441" />
        <edge x="288" y="287" />
        <edge x="256" y="278" />
        <edge x="236" y="243" />
        <edge x="185" y="234" />
    </border>
    <cities>
        <city name="Flowerville" x="305" y="109" type="ours">
            <sells>
                <resource type="vegetables"/>
                <resource type="iron"/>
            </sells>
        </city>
        <city name="Zargoville" x="550" y="50" trade_route_cost="750" trade_route_type="sea">
            <buys>
                <resource type="vegetables" amount="200"/>
                <resource type="marble" amount="230"/>
            </buys>
            <sells>
                <resource type="fruit" amount="16"/>
                <resource type="wheat" amount="190"/>
            </sells>
            <trade_points>
                <point x="350" y="112"/>
                <point x="447" y="51"/>
                <point x="480" y="40"/>
            </trade_points>
        </city>
        <city name="Biancaville" x="654" y="16" type="distant"/>
        <city name="Fairbuyville" x="340" y="210" type="vulnerable"/>
    </cities>
    <invasion_paths>
        <path>
            <battle x="290" y="300"/>
            <battle x="180" y="210"/>
            <battle x="170" y="130"/>
        </path>
        <path>
            <battle x="460" y="65"/>
            <battle x="340" y="70"/>
            <battle x="210" y="60"/>
        </path>
    </invasion_paths>
    <distant_battle_paths>
        <path type="roman" start_x="200" start_y="100">
            <waypoint num_months="5" x="250" y="200"/>
            <waypoint num_months="4" x="350" y="225"/>
        </path>
        <path type="enemy" start_x="500" start_y="425">
            <waypoint num_months="9" x="350" y="225"/>
        </path>
    </distant_battle_paths>
</empire>
```

Add a file called test_empire.xml with the contents above and place it in the new custom_empires folder.

To select the new empire, click Empire -> Choose custom empire from the top menu in the scenario editor.
![image](https://user-images.githubusercontent.com/2081059/199633365-4e8fb075-b77c-4990-9445-c098becc1ac6.png)

Once that's selected, you should see something like this:
![Example custom empire](https://user-images.githubusercontent.com/6518876/217560550-a5e62927-abaa-44eb-890e-f06d8074fe12.png)

Clicking the Refresh button or hitting F5 will reload the empire from your XML file so you can make changes quickly. The current map coordinates of the cursor are displayed in the top left corner. Objects are typically positioned based on their top left corner.

The XML file has the following settings:
- `<empire>` Opening tag. Version is always set to 1 at this point. This is a placeholder in case we decide to make a future XML empire file format that is incompatible.
- `<ornament>` Adds an ornament (a drawing) to the empire map. Its position is based on its type. It has a single required `type` attribute, which can be one of the following:
         - `The Stonehenge`
        - `Gallic Wheat`
        - `The Pyrenees`
        - `Iberian Aqueduct`
        - `Triumphal Arch`
        - `West Desert Wheat`
        - `Lighthouse of Alexandria`
        - `West Desert Palm Trees`
        - `Trade Ship`
        - `Waterside Palm Trees`
        - `Colosseum`
        - `The Alps`
        - `Roman Tree`
        - `Greek Mountain Range`
        - `The Parthenon`
        - `The Pyramids`
        - `The Hagia Sophia`
        - `East Desert Palm Trees`
        - `East Desert Wheat`
        - `Trade Camel`
- `<border>` Optional. The empire border. Possible attribute is:
    - `density` Optional. The distance, in pixels, between each empire border flag. Defaults to `50`.
- `<edge>` An edge of the border. You can add as many as you want to make the border more natural-looking. The game will add flags at specific intervals  on a straight line between each edge. The last edge of the border will always connect to the first. Possible attributes are:
    - `x` Required. X coordinate of the edge.
    - `y` Required. Y coordinate of the edge.
    -  `hidden`. Optional. If set to `true`, the line going from the current edge to the next won't show any flags. This allows you to not show the flags i.e. when the border crosses the ocean.
- `<cities>` The list of cities.
- `<city>` Defines a city. Possible attributes are:
    - `name` Required. What your city is called.
    - `x` Required. X coordinate of your city.
    - `y` Required. I'm going to let you figure this one out on your own.
    - `type` Optional. Defaults to `trade`. Possible values are:
        - `ours` This is the city you're making!
        - `roman` A distant Roman city that can't be traded with (blue flag on empire map).
        - `distant` A distant foreign city that 'can't be traded with (yellow flag on empire map).
        - `trade` Default setting. A city that can be traded with.
        - `vulnerable` A distant Roman city that can be attacked. Must be specified to support distant battles.
    - `trade_route_cost` Optional. For trade cities, how much it costs to open a trade route with them. Defaults to `500`.
    - `trade_route_type` Optional. Valid values are `land` or `sea`. Defaults to `land`.
    - `<buys>` and `<sells>` Opening tags for resources that a city will buy and sell, respectively. If the city `type` is set to `ours`, the `<sells>` list is required to specify what resources you can produce.
        - `<resource>` Tag for a resource that can be bought or sold. Attributes are:
            - `type` The resource type. Currently accepted values are:
                - `wheat`
                - `vegetables`
                - `fruit`
                - `olives`
                - `vines`
                - `meat`
                - `fish` 
                - `wine`
                - `oil`
                - `iron`
                - `gold`
                - `timber` or `wood`
                - `clay`
                - `marble`
                - `weapons`
                - `furniture`
                - `pottery`
        - `amount` How much this city is willing to trade. Can be left blank if city `type` is set to `ours`. Defaults to `1`. This amount does _not_ need to match the 0/15/25/40 values used in the base game.
    - `<trade_points>` Optional. Opening tag to define waypoints for your dotted trade path to take on the empire map. If not specified, a line will be drawn directly between the home city and the trade city. If specified, they go from your city in the direction of the trading city.
        - `<point>` Defines a trade path waypoint. Attributes are `x` and `y` for the position.
- `<invasion_paths>` This section defines the battle markers that appear on the map when enemies are closing in on your city. While it is optional, invaders WILL NOT be able to attack the city if there is not at least one path defined. Invaders will move up one battle marker for each year that passes until they eventually invade.
    - `<path>` Defines a new invasion path.
        - `<battle>` Defines an invasion battle marker. Attributes are `x` and `y`.
- `<distant_battle_paths>` Opening tag to define paths for friendly and enemy armies for city rescues.
    - `<path>` A path taken by an army to a distant city. Attributes are:
        - `type` Specifies the army type that uses this path. `roman` for troops from your city or `enemy` for invaders.
        - `start_x` Starting X coordinate for the path.
        - `start_y` Starting Y coordinate for the path.
        - `<waypoint>` Defines a path waypoint used to draw a path between the `start_x` and `start_y` or the last waypoint's `x` and `y` coordinate. Attributes are:
            - `num_months` The number of points along the path. It takes an army 1 month to travel to a new point.
            - `x` and `y` The coordinates for the end of this waypoint segment.

If you'd like to use some of the cities from the base game, here's a list of them compiled by @MSTVD with their coordinates:
```txt
Roma, 752, 504
Tarentum, 905, 558
Brundisium, 945, 543
Syracusae, 835, 733
Carthago Nova, 225, 728
Mediolanum, 614, 310
Tarraco, 382, 494
Valentia, 277, 617
Toletum, 114, 603
Tingis, 115, 794
Sinope, 1625, 491
Tarsus, 1657, 733
Leptis Magna, 835, 894
Cyrene, 1128, 883
Lutetia, 373, 191
Massilia, 473, 400
Narbo, 395, 426
Lugdunum, 473, 284
Caesarea, 382, 755
Alexandria, 1508, 929
Augusta Trevorum, 506, 125
Argentoratum, 552, 157
Volubilis, 61, 903
Lindum, 266, 16
Byzantium, 1380, 521
Sarmizegetusa, 1041, 307
Thamugadi, 579, 857
Calleva, 210, 77
Londinium, 322, 67
Antiocha, 1718, 755
Heliopolis, 1739, 785
Damascus, 1742, 815
Hierosolyma, 1698, 875
Pergamum, 1366, 613
Ephesus, 1337, 678
Miletus, 1353, 712
Athenae, 1172, 659
Corinthus, 1133, 681
Carthago, 656, 735
Capua, 802, 517
```

**Comment by Salve970** [comment]
Great Idea.  Did you use XML Notepad?  Thanks.

**Reply by PrettyFlower** [reply]
You can use whatever XML text editor you want. Personally I use visual studio code: https://code.visualstudio.com/

**Reply by Salve970** [reply]
Thanks

**Comment by Ouaz** [comment]
**- Opening post updated -**

For reference:

By Crudelios:

Empire borders have now been added. The code is simple enough:

```
<empire version="1">
    <border>
        <edge x="300" y="40" />
        <edge x="100" y="200" />
        <edge x="500" y="200" />
    </border>
        <!-- etc.. -->
```
The border will automatically close from the last point to the first. (If you don't want the border to be closed, add `hidden="true"` to the last `edge` tag )
If a flag is above water, add `hidden="true"` to the corresponding `edge` tag.

```
<border density="25">
        <edge x="169" y="215" hidden="true" />
        <edge x="142" y="157" />
        <edge x="181" y="126" />
        <edge x="217" y="128" />
        <edge x="233" y="110" />
        <edge x="312" y="71" />
        <edge x="325" y="78" />
        <edge x="344" y="70" />
        <edge x="370" y="84" hidden="true" />
        <edge x="454" y="126" />
        <edge x="496" y="282" />
        <edge x="466" y="359" />
        <edge x="474" y="425" />
        <edge x="465" y="433" />
        <edge x="440" y="419" />
        <edge x="409" y="444" />
        <edge x="402" y="469" />
        <edge x="262" y="441" />
        <edge x="288" y="287" />
        <edge x="256" y="278" />
        <edge x="236" y="243" />
        <edge x="185" y="234" />
    </border>
``` 

You can also set the distance between the border flags.
`border density="25"`
In that case it's 25 pixels, default is 50.

![custom_empire_borders](https://user-images.githubusercontent.com/11946570/217359450-0931fc4e-7642-4daa-b769-71981065826b.png)

**Reply by MaximusRink** [reply]
First off, thank you to everyone involved and who continues to be involved with this! C3 is the game that got me into Computer Empire Building games years ago. The Augustus Mod is so good it's borderline a sequel to C3. Is there way a that we can make a donation to the development team as a thank you?

Now...onto my question.
For some reason I cannot get the custom empire files to show up.
-I am running 4.0 on the GOG C3 version.
-I created a "custom_empires" folder in the following path -- C:\GOG Games\Caesar 3
-I copied and pasty the suggested text.
-Nothing shows up when I go to the custom empires tab in game.

Not sure what I'm doing wrong. While I am far from a programmer, I am semi-literate enough to modify graphics and build custom buildings and civilizations in Age of Empires 3 for those who know that game, so I'm not sure why I'm having a problem here. It's probably just some knuckleheaded oversite on my part, but any help would be appreciated.

Thanks in advance!
![C1](https://github.com/user-attachments/assets/ee626f52-4345-4220-b047-bb2c0b0c3c7e)
![C2](https://github.com/user-attachments/assets/ea103d99-3dda-4221-8860-cc59eec06a0f)
![C33](https://github.com/user-attachments/assets/8cc8c02e-b5d6-43c3-b802-4250ac74749a)

-

**Reply by crudelios** [reply]
Hey, thanks for the kind words!

Don't worry about donating, we do this for fun and if you enjoy the game, that's gratification enough for us 🙂 

Regarding your issue: are you using the v4.0.0 release or the latest unstable? I ask because the directories changed.

You can check your version on the main menu!

**Reply by MaximusRink** [reply]
the v4.0.0 release I believe. Here's a screen shot.
![C5](https://github.com/user-attachments/assets/84cea537-8b16-4be8-bca2-e7e6ec2bf75a)

**Reply by crudelios** [reply]
That should be the right directory indeed.

I'm guessing the "open file" window appears but it's empty, right?

The only issue I see is that the file name is `test.xml.xml`. It should be only `test.xml`.

**Reply by MaximusRink** [reply]
That was it! Made my custom empire with 7 sea and 7 land trade routes. Tested everything. Works like a charm! Many many thanks for the speedy help and, once again, the entire Augustus update!

**Comment by Ouaz** [comment]
For reference:

Full list of ornaments:

   ```
 "The Stonehenge",
    "Gallic Wheat",
    "The Pyrenees",
    "Iberian Aqueduct",
    "Triumphal Arch",
    "West Desert Wheat",
    "Lighthouse of Alexandria",
    "West Desert Palm Trees",
    "Trade Ship",
    "Waterside Palm Trees",
    "Colosseum|The Colosseum",
    "The Alps",
    "Roman Tree",
    "Greek Mountain Range",
    "The Parthenon",
    "The Pyramids",
    "The Hagia Sophia",
    "East Desert Palm Trees",
    "East Desert Wheat",
    "Trade Camel",
    "Mount Etna",
    "Colossus of Rhodes"
```

All ornaments can be added at once (no need to type the full list) with:

```
<ornament>
          type="all"
</ornament>
```

Ireland can at last be shown on the empire. Add `show_ireland="true"`, to the empire tag:

`<empire version="1" show_ireland="true">` (...etc)

**Reply by BeHo86** [reply]
This is amazing! Is the old limit of max 4 tradable cities still active or can we have more in the meantime? Especially huge cities of some tens of thousands of inhabitants would really benefit from increased trade imho

**Reply by Ouaz** [reply]
I didn't test, but I think the limit has been removed.

**Reply by crudelios** [reply]
Yes there are no limits to the number of tradeable cities on custom empires.

**Reply by BeHo86** [reply]
Thanks for the confirmation, the feature is much appreciated!
However, custom empires don't work for me - I made the folder in my C3 base / installation directory and placed the test_empire.xml file there. However, in the (Augustus) map editor, the option "Select custom empire" didn't show up (Just "Select location").
I'm using Augustus 3.2.0 (latest stable) with the German version of the game (if that matters).

EDIT: just realized I need to use the latest unstable build. All working now

**Comment by XLightningStormL** [comment]
Due to latter changes, all of the Vanilla cities from MSTVD's coordinates have to be adjusted by +20 X and Y. With Basic XML format here it is provided below, it is up to you to modify them for distant cities or trade cities:

```
	<city name="Roma" x="772" y="524" type="roman"/>
        <city name="Tarentum" x="925" y="578" type="roman"/>
        <city name="Brundisium" x="965" y="563" type="roman"/>
        <city name="Syracusae" x="855" y="753" type="roman"/>
        <city name="Carthago Nova" x="245" y="748" type="roman"/>
        <city name="Mediolanium" x="634" y="330" type="roman"/>
        <city name="Tarraco" x="402" y="514" type="roman"/>
        <city name="Valentia" x="297" y="637" type="roman"/>
        <city name="Toletum" x="134" y="623" type="roman"/>
        <city name="Tingis" x="135" y="814" type="roman"/>
        <city name="Sinope" x="1645" y="511" type="roman"/>
        <city name="Tarsus" x="1677" y="753" type="roman"/>
        <city name="Leptis Magna" x="855" y="914" type="roman"/>
        <city name="Cyrene" x="1148" y="903" type="roman"/>
        <city name="Lutetia" x="393" y="211" type="roman"/>
        <city name="Massilia" x="493" y="420" type="roman"/>
        <city name="Narbo" x="415" y="446" type="roman"/>
        <city name="Lugdunumn" x="493" y="304" type="roman"/>
        <city name="Caesaerea" x="402" y="775" type="roman"/>
        <city name="Alexandria" x="1528" y="949" type="roman"/>
        <city name="Augusta Trevorum" x="526" y="145" type="roman"/>
        <city name="Argentoratum" x="572" y="172" type="roman"/>
        <city name="Volubilis" x="81" y="923" type="roman"/>
        <city name="Lindum" x="286" y="36" type="roman"/>
        <city name="Byzantium" x="1400" y="541" type="roman"/>
        <city name="Sarmizegetusa" x="1061" y="327" type="roman"/> <!--Actually Aquincum-->
        <city name="Thamugadi" x="599" y="877" type="roman"/>
        <city name="Calleva" x="230" y="97" type="roman"/>
        <city name="Londinium" x="342" y="87" type="roman"/>
        <city name="Antiocha" x="1738" y="775" type="roman"/>
        <city name="Heliopolis" x="1759" y="805" type="roman"/> <!--Actually Palmyra-->
        <city name="Damascus" x="1762" y="815" type="roman"/>
        <city name="Hierosolyma" x="1718" y="895" type="roman"/>
        <city name="Pergamum" x="1386" y="633" type="roman"/>
        <city name="Epheseus" x="1357" y="698" type="roman"/>
        <city name="Miletus" x="1373" y="738" type="roman"/>
        <city name="Athenae" x="1192" y="679" type="roman"/>
        <city name="Corinthus" x="1153" y="701" type="roman"/>
        <city name="Carthago" x="676" y="755" type="roman"/>
        <city name="Capua" x="822" y="537" type="roman"/>
```

As a bonus, I have also provided additional cities relevant to the romans, or the modern day, including alternatives to some of the inaccurate C3 Cities:

```
        <city name="Arretium" x="746" y="444" type="roman"/>
        <city name="Neapolis" x="825" y="552" type="roman"/> <!--Not Compatible with Capua-->
        <city name="Pompeii" x="870" y="592" type="roman"/>
        <city name="Arminium" x="772" y="407" type="roman"/>
        <city name="Aquileia" x="765" y="295" type="roman"/>
        <city name="Caralis" x="625" y="640" type="roman"/>
        <city name="Gades" x="102" y="755" type="roman"/>
        <city name="Dyrrhachium" x="1040" y="560" type="roman"/>
        <city name="Nicopolis" x="1094" y="662" type="roman"/>
        <city name="Salonae" x="890" y="410" type="roman"/>
        <city name="Messene" x="870" y="711" type="roman"/>
        <city name="Lilybaeum" x="756" y="720" type="roman"/>
        <city name="Aquincum" x="1061" y="327" type="roman"/><!--Sarmizegetusa in the vanilla game-->
        <!--Alternate--><city name="Sarmizegetusa" x="1194" y="369" type="roman"/> <!--The real one-->
        <city name="Thessalonica" x="1202" y="590" type="roman"/>
        <city name="Chersonesus" x="1607" y="362" type="roman"/>
        <city name="Artaxata" x="1930" y="595" type="roman"/>
        <city name="Trapezus" x="1840" y="570" type="roman"/>
        <city name="Petra" x="1673" y="972" type="roman"/>
        <city name="Eboracum" x="265" y="0" type="roman"/>
        <city name="Lauriacum" x="785" y="208" type="roman"/>
        <city name="Ctesiphon" x="1885" y="850" type="roman"/>
        <city name="Babylon" x="1895" y="895" type="roman"/>
        <!--Alternate--><city name="Heliopolis" x="1554" y="974" type="roman"/> <!--The real Heliopolis in Egypt, just rename the old one to Palmyra-->
```

**Comment by UberWaffe** [comment]
Example of a basic custom empire. This one is a variant on the Meliodanum empire map.
```
<?xml version="1.0"?>
<!DOCTYPE empire>
<empire version="1">
    <cities>
        <city name="Mediolanum" x="614" y="310" type="ours">
            <sells>
                <resource type="wheat"/>
                <resource type="vegetables"/>
                <resource type="meat"/>
                <resource type="olives"/>
                <resource type="fish"/>
                <resource type="iron"/>
                <resource type="timber"/>
                <resource type="clay"/>
                <resource type="wine"/>
                <resource type="oil"/>
                <resource type="weapons"/>
                <resource type="furniture"/>
                <resource type="pottery"/>
            </sells>
        </city>
        <city name="Capua" x="802" y="517" trade_route_cost="925">
            <buys>
                <resource type="wine" amount="15"/>
                <resource type="weapons" amount="15"/>
                <resource type="furniture" amount="15"/>
            </buys>
            <sells>
                <resource type="oil" amount="25"/>
                <resource type="iron" amount="25"/>
                <resource type="timber" amount="25"/>
            </sells>
            <trade_points>
                <point x="632" y="383"/>
                <point x="671" y="403"/>
                <point x="683" y="452"/>
                <point x="710" y="482"/>
                <point x="782" y="541"/>
            </trade_points>
        </city>
        <city name="Narbo" x="395" y="426" trade_route_cost="950">
            <buys>
                <resource type="timber" amount="25"/>
                <resource type="furniture" amount="15"/>
                <resource type="pottery" amount="15"/>
            </buys>
            <sells>
                <resource type="olives" amount="25"/>
                <resource type="vines" amount="25"/>
                <resource type="wine" amount="15"/>
            </sells>
            <trade_points>
                <point x="518" y="432"/>
                <point x="498" y="422"/>
                <point x="456" y="380"/>
                <point x="418" y="407"/>
            </trade_points>
        </city>
        <city name="Massilia" x="473" y="400" trade_route_cost="800">
            <buys>
                <resource type="wheat" amount="60"/>
                <resource type="furniture" amount="15"/>
                <resource type="pottery" amount="30"/>
            </buys>
            <sells>
                <resource type="wine" amount="15"/>
                <resource type="oil" amount="15"/>
                <resource type="timber" amount="25"/>
            </sells>
            <trade_points>
                <point x="608" y="374"/>
                <point x="557" y="410"/>
                <point x="540" y="418"/>
                <point x="505" y="445"/>
            </trade_points>
        </city>
        <city name="Tarentum" x="905" y="558" trade_route_cost="1275">
            <buys>
                <resource type="vegetables" amount="40"/>
                <resource type="timber" amount="25"/>
                <resource type="furniture" amount="15"/>
            </buys>
            <sells>
                <resource type="olives" amount="25"/>
                <resource type="wine" amount="25"/>
                <resource type="oil" amount="25"/>
            </sells>
            <trade_points>
                <point x="720" y="377"/>
                <point x="802" y="435"/>
                <point x="813" y="492"/>
            </trade_points>
        </city>
        <city name="Syracusae" x="835" y="733" trade_route_cost="1900" trade_route_type="sea">
            <buys>
                <resource type="vegetables" amount="25"/>
                <resource type="iron" amount="15"/>
                <resource type="weapons" amount="15"/>
            </buys>
            <sells>
                <resource type="vines" amount="25"/>
                <resource type="marble" amount="15"/>
                <resource type="pottery" amount="25"/>
                <resource type="fruit" amount="120"/>
            </sells>
            <trade_points>
                <point x="660" y="475"/>
                <point x="740" y="539"/>
                <point x="881" y="632"/>
                <point x="883" y="698"/>
                <point x="879" y="729"/>
            </trade_points>
        </city>
        <city name="Tingis" x="115" y="794" trade_route_cost="1800" trade_route_type="sea">
            <buys>
                <resource type="wheat" amount="300"/>
                <resource type="vegetables" amount="80"/>
                <resource type="pottery" amount="80"/>
                <resource type="oil" amount="60"/>
            </buys>
            <sells>
                <resource type="fruit" amount="80"/>
                <resource type="wine" amount="60"/>
            </sells>
            <trade_points>
                <point x="587" y="412"/>
                <point x="528" y="455"/>
                <point x="474" y="456"/>
                <point x="432" y="485"/>
                <point x="412" y="544"/>
                <point x="345" y="581"/>
                <point x="308" y="734"/>
                <point x="290" y="747"/>
                <point x="259" y="781"/>
            </trade_points>
        </city>
        <city name="Tarraco" x="382" y="494" type="vulnerable"/>
        <city name="Toletum" x="114" y="603" type="roman"/>
        <city name="Carthago Nova" x="225" y="728" type="roman"/>
        <city name="Valentia" x="277" y="617" type="roman"/>
        <city name="Brundisium" x="945" y="543" type="roman"/>
        <city name="Corinthus" x="1133" y="681" type="roman"/>
        <city name="Athenae" x="1172" y="659" type="roman"/>
        <city name="Pergamum" x="1366" y="613" type="roman"/>
        <city name="Ephesus" x="1337" y="678" type="roman"/>
        <city name="Miletus" x="1353" y="712" type="roman"/>
        <city name="Carthago" x="656" y="735" type="distant"/>
        <city name="Leptis Magna" x="835" y="894" type="distant"/>
        <city name="Cyrene" x="1128" y="883" type="distant"/>
        <city name="Byzantium" x="1380" y="521" type="distant"/>
        <city name="Sinope" x="1625" y="491" type="distant"/>
        <city name="Tarsus" x="1657" y="733" type="distant"/>
    </cities>
    <invasion_paths>
        <path>
            <battle x="570" y="304"/>
            <battle x="544" y="275"/>
            <battle x="460" y="200"/>
        </path>
        <path>
            <battle x="645" y="313"/>
            <battle x="644" y="253"/>
            <battle x="660" y="180"/>
        </path>
    </invasion_paths>
    <distant_battle_paths>
        <path type="roman" start_x="593" start_y="321">
            <waypoint num_months="4" x="420" y="368"/>
            <waypoint num_months="4" x="350" y="483"/>
            <waypoint num_months="2" x="380" y="540"/>
        </path>
        <path type="enemy" start_x="300" start_y="260">
            <waypoint num_months="8" x="324" y="531"/>
            <waypoint num_months="2" x="380" y="540"/>
        </path>
    </distant_battle_paths>
</empire>
```

**Reply by UberWaffe** [reply]
If you want to change what raw resources your own city can produce (farms, mines and quarries) then you can do so by changing the city that is "ours".
```
        <city name="MyRome" x="614" y="310" type="ours">
            <sells>
                <resource type="wheat"/>
                <resource type="olives"/>
                <resource type="clay"/>
            </sells>
        </city>
```
In the above example, the city called "MyRome" is set to be "ours", so it is the city on the map we will play as.
It will be able to farm wheat and oilves. It will be able to build clay pits.

Edit: Fixed example as per note from @crudelios

**Reply by crudelios** [reply]
Just a small errata: this `sells` approach only applies to raw materials.

For finished goods, the game checks if raw materials can be produced or imported and, if that's the case, then the workshop is automatically enabled.

Maybe we should replace the `sells` tag with a `raw_materials` for our city.

**Comment by f3nics** [comment]
If I use a custom empire, I cannot build monuments (colosseum, hippodrome, etc), the button does not exist anymore! Tradings worked. Is it a bug?
Thanks for any help

**Reply by crudelios** [reply]
Old custom empires aren't updated to support the new resources, so the monuments can't be built.

You should ask the original map maker, if possible, to update his map.

**Reply by f3nics** [reply]
You are right! Thanks

**Comment by AlStarUA** [comment]
Hello.
I am creating a new map for a game with a custom empire. But there is no possibility to build any monuments in the new map.
Does the custom empire have support for monuments and, if so, how do I enable them? From the comments above, I did not understand whether there is support for building monuments in a custom empire or whether I need to wait for updates.
Augustus version: 3.2.0.550 (last build)

**Reply by crudelios** [reply]
Hi,

There is support for the new monuments but you need to enable the new resources, either by allowing local production or by importing them.

Monuments that don't have access to the require resources can't be constructed, so they won't even appear on the build menu.

**Reply by AlStarUA** [reply]
Thank you very much.

I completely forgot about the new resources!

**Comment by badgermolegit** [comment]
I've followed the instructions to add a custom_empires folder with a test_empire.xml file but nothing comes up when I try to open a custom empire. 

I have the latest Augustus unstable build. I downloaded the editor from Julius. I'm unsure what I am doing wrong. 

I'm also unable to find how to place fake rocks (so visually rocks are present but are not mine-able), and I wonder if it is related. Thanks for any help!

![Screenshot 2025-02-18 164155](https://github.com/user-attachments/assets/e31a72a7-9539-4946-b790-7d73240cefcd)

**Comment by Ouaz** [comment]
> I've followed the instructions to add a custom_empires folder with a test_empire.xml file but nothing comes up when I try to open a custom empire.

Ah, yes, it's because the folder structure has changed in the latest unstable builds.

Add a folder named `editor` in your Caesar 3 folder, then a subfolder named `empires`, then put your xml file inside:

`editor > empires > yourfile.xml`

You can also use the "`Set user directory`" option in "General settings" menu, and choose "`Caesar III installation path, using sub-directories`", so Augustus creates automatically all the required folders.

**Reply by badgermolegit** [reply]
This isn't working either - 

Went into General settings, changed to "Caesar III installation path, using sub-directories" (was on Default before)

Added xml here: 
![Screenshot 2025-02-19 112017](https://github.com/user-attachments/assets/15312ff8-eff5-4db6-a6de-1e6d51202197)

still isn't showing up when I go into the editor and try to load a custom empire. There is nothing in the list. 

![image](https://github.com/user-attachments/assets/3887a272-7ac2-4d4e-a0b7-02cb6a16152b)

**Reply by badgermolegit** [reply]
I tried a fresh install, set the directory to Caesar 3 path using sub-directories, added the test_empire.xml to the editor > empire folder, and it isn't working again. 

The reconquered campaign also doesn't seem to exist in the fresh install, which I find odd.

**Reply by Ouaz** [reply]
Copy this simple empire.xml file in the empires folder. Does it appear in the editor? 
[crudelville.zip](https://github.com/user-attachments/files/18874090/crudelville.zip)

> The reconquered campaign also doesn't seem to exist in the fresh install, which I find odd.

What do you mean?
Augustus build 544 creates a folder named `campaigns`, and you have to unzip `Reconquered Campaign` folder ([the new one](https://caesar3.heavengames.com/downloads/showfile.php?fileid=2243)) in it like this: `campaigns > Reconquered Campaign` (which contains `audio`, `image`, `scenario`, etc. folders)

**Reply by badgermolegit** [reply]
The crudelville file worked! So it appears my file wasn't saving properly in xml format, it was just saving as a text file with .xml added, I see now. I should have caught that! 

I thought the RC came with the Augustus unstable build files, my apologies. I can't thank you enough, I've been itching to create some custom maps. Thank you!

**Comment by Filu2259** [comment]
I've created a google sheet that you simply fill out to obtain your empire ready to paste into wordpad and save as .xml

As a user, you can generate a fully functional empire in 2 minutes using your web browser. Just fill out cells that light up as you progress. 

Use the "create your own empire here" sheet directly from the link, or duplicate it , or create a private copy in your own caesar 3 speadsheet. Once you are done you may leave your creation available to others.

It focuses mostly on generating cities and trade quota, but also includes rudimentary tools to create military paths, tradepoints, and borders, as shown in the example sheet.

It may or may not receive improvements in the future. Please let me know if you encounter any issues using it or if you volunteer to teach me how to use speadsheets correctly.

https://docs.google.com/spreadsheets/d/1kDKDmPKDzQ-CmzT01WKSbe1aZO0auZtEjrtuAEXmGus/edit?usp=sharing

**Reply by Filu2259** [reply]
few fixes, cleanups, and improvements.

**Comment by Salve970** [comment]
How do you get Custom Empire to work in Augustus?  I tried instructions above but get error message saying unable yo load.

**Reply by Johannes-callidus** [reply]
Can you upload your empire here please. I'll probably be able to fix it.

**Reply by Salve970** [reply]
Ok....Thanks....OK..Thanks.....
[test_empire.xml](https://github.com/user-attachments/files/23822994/test_empire.xml)

**Comment by Salve970** [comment]
OK..Thanks.....
[test_empire.xml](https://github.com/user-attachments/files/23822994/test_empire.xml)

**Reply by Filu2259** [reply]
<resource type="gold"></resource>
        <resource type="sand"></resource>
        <resource type="stone"></resource>
        <resource type="bricks"></resource>
      </sells>

**Reply by Filu2259** [reply]
you have typos in the resources from your own city

**Reply by Filu2259** [reply]
our city doesn't need "buys" and your "sells" have these 4 unnecessary resource tags.

**Reply by Johannes-callidus** [reply]
There also were some other problems.

**Reply by Johannes-callidus** [reply]
It still returns an error though.

**Reply by Filu2259** [reply]
if he used my spreadsheet it would work n_n

**Reply by Johannes-callidus** [reply]
Maybe, can't tell. This is the document with a lot of fixes but it still doesn't load. My debbuger says that a line doesn't start with a '<' but I think that's wrong. Though I believe in the xml parser as it's external. If that's the problem though which I don't think we can't do anything.
[test_empire.xml](https://github.com/user-attachments/files/23824932/test_empire.xml)

**Comment by Johannes-callidus** [comment]
What did you use to create this file and which encoding?

**Reply by Salve970** [reply]
I used Notepad

**Reply by Johannes-callidus** [reply]
Encoding, something like UTF-8 or ISO-8859 or WINDOWS-1250?

**Reply by Salve970** [reply]
Windows. Thanks for fast response and helping.

**Reply by Johannes-callidus** [reply]
Please use UTF-8 in the future the empire has some characters which the xml_parser didn't understand as they aren't in UTF-8 (WINDOWS 1252 has 256 different characters and UTF-8 only 128). I'm unfortunatly not sure how to fix it other then rewriting the whole file or atleast most of it. Also I'm not a hundred percent sure whether that was the whole cause of the problem.

**Reply by Salve970** [reply]
Thanks for feedback. How do you get UTF-8 ?

**Reply by Filu2259** [reply]
FYI, I openend the test_empire.xml shared by Johannes, copy, paste into a blocnote, save with the .xml extension and encoding AINSI (its a field you can choose below the field where you write the name.xml. utf 8 is also available there. 

it worked in the map editor no problem.

**Reply by Salve970** [reply]
Do you have a free download of Blocnote?

**Comment by Johannes-callidus** [comment]
[test_empire.xml](https://github.com/user-attachments/files/23835063/test_empire.xml)
Here's your empire in UTF-8 ready to be loaded in the map editor.

**Reply by Salve970** [reply]
Thank you. It works.

**Comment by Johannes-callidus** [comment]
If you have other files in the wrong encoding you can just use an online encoding converter to convert them to UTF-8.

**Reply by Salve970** [reply]
Thanks

**Comment by Filu2259** [comment]
roman and enemy distant battle paths must imperatively total more than 5 months each.

Otherwise the request will not happen. Thanks Eugenikus for finding this in the code!
